### PR TITLE
Add detail command

### LINKED ISSUE #150: Detail Command
- Allow for users to view contacts in detail

- [x] Change UI
- [x] Add Detail Command Classes

### PR DESCRIPTION
**Pull request by Angelyxx**
Allows users to view more details of a contact chosen by the displayed index.

Details are shown by pulling attributes from a Person Object and displaying them onto PersonViewPanel.

Displays "Welcome to Connexion!" if there are no existing contacts, else defaults to the first contact in the list upon initialisation. 

Updates to existing Person data through commands (e.g. Mark, edit, etc) are not instantly reflected onto the PersonViewPanel, changes to this UI can only be done via the DetailCommand as intended.

Shifted Schedule to a row above Note (in comparison to our mock-up) because it's inconvenient if user has a long note for the contact.

CodeCov report looks terrible 🤡 because this PR has heavy UI changes

_Notes for future implementation on:_

 **ScheduleCommand** (@geoffong11, @xavierpok)
-  `getDetailString()` for the Schedule class should follow "Upcoming Meeting: " + (string representation of the schedule)
-   `schedule.setText("Upcoming Meeting: 10 Oct (in 1 year)");` is currently set in the PersonViewPanel as a dummy string representation for schedule (delete when no longer needed)
- `//schedule.setText(person.getSchedule().getDetailString());` uncomment/edit this line in PersonViewPanel when done!

 **NoteCommand** (@imkwokyong)
-  `getDetailString()` for the Note class would just be the text of the note itself (as note is likely to be multi-lined, hence the fxml itself has a seperate header for notes)
-   `note.setText("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam volutpat, quam ac vehicula.");` is currently set in the PersonViewPanel as a dummy string representation for note (delete when no longer needed)
- `//note.setText(person.getNote().getDetailString());` uncomment/edit this line in PersonViewPanel when done!

closes #150 

Passes all test cases.

_Screenshots of the UI:_
After running Detail Command:
Screenshot 2023-10-26 at 4 39 24 AM

No existing contacts:
Screenshot 2023-10-26 at 4 40 58 AM

### CODE REVIEW

**Review by imkwokyong — COMMENTED** [review]
There are some bugs that I found after tinkering with the code:
1. Tags is not in personViewPanel
2. `mark` resets to false after `edit`
3. Lastmodifieddatetime is supposed to be in personViewPanel, not personListPanel

Suggestion for improvements:
1. if said person in personViewPanel is edited or marked, can show the new edits/mark immediately on the personViewPanel, without having to type in the detail command again
2. If said person in personViewPanel is deleted, clear the personViewPanel too.

**Review by imkwokyong — APPROVED** [review]
LGTM, just some nits like the suggestions I mentioned earlier. Can try to do it if you have time.
For the moving of the LastModifiedDateTime, I will try to do it in my PR.

**Comment by Angelyxx** [comment]
> Suggestion for improvements: (can do this if you have time)
> 
> 1. if said person in personViewPanel is edited or marked, can show the new edits/mark immediately on the personViewPanel, without having to type in the detail command again
> 2. If said person in personViewPanel is deleted, clear the personViewPanel too.

Attempted this and it worked, however it failed tests (e.g. DeleteCommandTest) as it lead to a MainWindow null pointer due to the call to UiManager when running test cases, which I'm not sure how to resolve :(

For now, I've comment out the line responsible for UI change `//UiManager.updatePersonView();` in the classes for DeleteCommand, EditCommand, MarkCommand and Unmark Command.

**Review by imkwokyong — APPROVED** [review]
LGTM!

**Comment by xavierpok** [comment]
After looking at codecov report & taking into account parts that were being tested, test coverage is OK.

Big areas of missing test coverage mostly concern the entire UI which was mostly untested before.

File: src/test/java/connexion/logic/commands/DetailCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+public class DetailCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        DetailCommand detailCommand = new DetailCommand(outOfBoundIndex);
+        assertCommandFailure(detailCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
```
Comment (xavierpok) [inline src/test/java/connexion/logic/commands/DetailCommandTest.java:29]: While it's not _absolutely_ needed, I'm not that comfortable with the test coverage here. I'm mostly referring to the precedent set by `DeleteCommandTest` where we test all combinations of (filtered list, unfiltered list) x (valid, invalid)

I think it's okay overall after some quick manual testing by launching and trying to call `detail` with invalid values, so this shouldn't stop us from merging.

**Review by xavierpok — APPROVED** [review]
Apart from some nits & minor issues, LGTM

**Pull request merged**